Add notimestamp linter (#105)
kubernetes-sigs/kube-api-linter: docs/linters.md

Maps are discouraged apart from `map[string]string` which is used for labels and annotations in Kubernetes APIs since it's hard to distinguish between structs and maps in spec. Instead of plain map, lists of named subobjects are preferred.

## Notimestamp

> 
> The `notimestamp` linter checks that the fields in the API types don't contain a 'Timestamp', or any field which contains 'Timestamp' as a substring, e.g CreateTimestamp.

Reviewer: Maybe?
```suggestion

The `notimestamp` linter checks that the fields in the API are not named with the word 'Timestamp'.
```
What is the reason for this from the API conventions, might be good to help explain that context here a little?

Author: Is there any other place apart from https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#naming-conventions, To look for reasons for avoiding it.

Author: sure thanks, I will check
